x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryStore.java

```diff
@@ -172,71 +168,38 @@ public void putAsync(ILMHistoryItem item) {
     }
 
     /**
-     * Checks if the ILM history index exists, and if not, creates it.
+     * Checks if the ILM history data stream exists, and if not, creates it.
      *
-     * @param client  The client to use to create the index if needed
+     * @param client  The client to use to create the data stream if needed
      * @param state   The current cluster state, to determine if the alias exists
-     * @param listener Called after the index has been created. `onResponse` called with `true` if the index was created,
+     * @param listener Called after the data stream has been created. `onResponse` called with `true` if the data stream was created,
      *                `false` if it already existed.
      */
-    @SuppressWarnings("unchecked")
-    static void ensureHistoryIndex(Client client, ClusterState state, ActionListener<Boolean> listener) {
-        final String initialHistoryIndexName = ILM_HISTORY_INDEX_PREFIX + "000001";
-        final IndexAbstraction ilmHistory = state.metadata().getIndicesLookup().get(ILM_HISTORY_ALIAS);
-        final IndexAbstraction initialHistoryIndex = state.metadata().getIndicesLookup().get(initialHistoryIndexName);
-
-        if (ilmHistory == null && initialHistoryIndex == null) {
-            // No alias or index exists with the expected names, so create the index with appropriate alias
-            logger.debug("creating ILM history index [{}]", initialHistoryIndexName);
-
-            // Template below should be already defined as real index template but it can be deleted. To avoid race condition with its
-            // recreation we apply settings and mappings ourselves
-            byte[] templateBytes = TEMPLATE_ILM_HISTORY.loadBytes();
-            Map<String, Object> templateAsMap = XContentHelper.convertToMap(new BytesArray(templateBytes, 0, templateBytes.length),
-                false, XContentType.JSON).v2();
-
-            client.admin().indices().prepareCreate(initialHistoryIndexName)
-                .setSettings((Map<String, ?>) templateAsMap.get("settings"))
-                .setMapping((Map<String, Object>) templateAsMap.get("mappings"))
-                .setWaitForActiveShards(1)
-                .addAlias(new Alias(ILM_HISTORY_ALIAS).writeIndex(true).isHidden(true))
-                .execute(new ActionListener<>() {
+    static void ensureHistoryDataStream(Client client, ClusterState state, ActionListener<Boolean> listener) {
```

Comment:
So I think we should remove this entirely, part of the reason we have data streams now is to not have to pre set up anything before indexing. Now that we wouldn't use aliases we don't have the complicated alias setup that requires us checking this every time.